libcontainer/cgroups/systemd/unified_hierarchy.go

```diff
@@ -209,10 +218,18 @@ func getv2Path(c *configs.Cgroup) (string, error) {
 		return "", err
 	}
 
+	if rootless {
+		uid, err := detectUID()
+		if err != nil {
+			return "", err
+		}
+		slice = fmt.Sprintf("user.slice/user-%d.slice/user@%d.service/%s", uid, uid, slice)
```

Comment:
You can also query this path from the D-Bus interface:

```
$ busctl --user get-property org.freedesktop.systemd1 /org/freedesktop/systemd1 org.freedesktop.systemd1.Manager ControlGroup
s "/user.slice/user-1001.slice/user@1001.service"
```

I think that's better, having systemd resolve that for you. Also future-proof if this path ever changes. (It has changed in the past.)

Reply:
Thanks, updated